testdata/xml-data/circular-references/analyse_circles_in_xml.py

```diff
@@ -30,19 +30,18 @@ def analyse_circles_in_data(xml_filepath: Path, tracer_output_file: str, save_tr
         max_stack_depth=3,
     )
     tracer.start()
-    tree = etree.parse(xml_filepath)
-    root = tree.getroot()
+    root = parse_and_clean_xml_file(xml_filepath)
```

Comment:
This function transforms the special tags <annotation>, <region>, and <link> to their technically correct form <resource restype="Annotation">, <resource restype="Region">, and <resource restype="LinkObj">. This is necessary, otherwise these special tags aren't considered when analyzing the links